metis-common/src/main/java/eu/europeana/metis/mongo/RecordRedirectDao.java

```diff
@@ -21,28 +21,40 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(RecordRedirectDao.class);
   private static final String OLD_ID = "oldId";
   private static final String NEW_ID = "newId";
+  private Datastore datastore;
 
-  protected MongoClient mongoClient;
-  protected Datastore datastore;
+  /**
+   * Constructor to initialize the mongo mappings/collections and the {@link Datastore} connection.
+   * This constructor is meant to be used when the database is already available.
+   *
+   * @param mongoClient the mongo client connection
+   * @param databaseName the database name of the record redirect database
+   */
+  public RecordRedirectDao(MongoClient mongoClient, String databaseName) {
+    this(mongoClient, databaseName, false);
+  }
 
   /**
-   * Constructs the DAO
-   * <p>Initializes the datastore using the provided mongo client</p>
+   * Constructor to initialize the mongo mappings/collections and the {@link Datastore} connection.
+   * This constructor is meant to be used mostly for when the creation of the database is required.
    *
    * @param mongoClient the mongo client connection
    * @param databaseName the database name of the record redirect database
-   * @param createIndexes flag that enables/disables the creation of the indexes in the database
+   * @param createIndexes flag that initiates the database/indices
    */
   public RecordRedirectDao(MongoClient mongoClient, String databaseName, boolean createIndexes) {
```

Comment:
I can see that the constructor with the createIndexes parameter is still called from the metis-redirects-migration project. Shouldn't that be changed?

Reply:
It should be changed.